assets/js/components/KeyMetrics/ChipTabGroup/index.js

```diff
@@ -115,16 +128,31 @@ export default function ChipTabGroup( { allMetricItems, savedItemSlugs } ) {
 		[ hasGeneratingLeadsGroup, hasSellingProductsGroup ]
 	);
 
+	const dynamicGroups = useMemo( () => {
+		if ( isUserInputCompleted ) {
+			return [ currentSelectionGroup, suggestedGroup ];
+		}
+		return [ currentSelectionGroup ];
+	}, [ isUserInputCompleted ] );
+
 	const allGroups = useMemo(
-		() => [ currentSelectionGroup, ...keyMetricsGroups ],
-		[ keyMetricsGroups ]
+		() => [ ...dynamicGroups, ...keyMetricsGroups ],
+		[ dynamicGroups, keyMetricsGroups ]
 	);
 
 	// Currently selected group does not include total selected number, so it will
 	// always be 0.
 	const selectedCounts = { [ KEY_METRICS_CURRENT_SELECTION_GROUP_SLUG ]: 0 };
 	const activeMetricItems = {};
 
+	if ( isActive === suggestedGroup.SLUG ) {
+		Object.keys( allMetricItems ).forEach( ( slug ) => {
+			if ( answerBasedMetrics.includes( slug ) ) {
+				activeMetricItems[ slug ] = allMetricItems[ slug ];
+			}
+		} );
+	}
```

Comment:
we already have this loop included bellow, no need to make another loop over the same items, we can move this code block bellow this check:
```js
if (
			metricGroup === isActive ||
			( isActive === currentSelectionGroup.SLUG &&
				effectiveSelection.includes( metricItemSlug ) )
		) {...}
```

So with a bit of adaptation/refactoring your code block can be included as:
```js
if ( isActive === suggestedGroup.SLUG && answerBasedMetrics.includes( metricItemSlug ) ) {
     activeMetricItems[ metricItemSlug ] =
				allMetricItems[ metricItemSlug ];
}
```

Reply:
Thank you, @zutigrm , this has been refactored in my recent commits.